tests/extensions/test_ormar_extension.py

```diff
@@ -33,14 +34,36 @@ class Meta(BaseMeta):
         pass
 
 
+class Job(Model):
+    id: int = Integer(autoincrement=True, primary_key=True)
+    person: Optional[Union[Person, Dict]] = ForeignKey(Person)
```

Comment:
This line is causing the issue for you. I am not sure why. Is person actually an Ormar model?

Reply:
The issue is different. `pydantic_factories` does not handle relations out of the box, at least in case of `OrmarModelFactory`.

See, if I do

```python
class JobFactory(OrmarModelFactory):
    __model__ = Job
    person = None
```

the test will pass, and the relation will be `None`. BUT it will also pass if I do:

```python
class JobFactory(OrmarModelFactory):
    __model__ = Job
    person = PersonFactory.build()
```

In that case a new Person instance will be generated each time `JobFactory` is used.

I think the behavior from the first snippet should be the default one for `pydantic_factories` when it comes to relations? At least for `OrmarModalFactory`? But maybe I'm wrong and it is desired to have to explicitly set it to either `None` or something else?

Reply:
Thats clear, but the error is that an int is set according to the error, no?

Eitherway, this should be fixed I guess. Also, why are you typing as optional dict or person? Why not simply person?

Reply:
> Thats clear, but the error is that an int is set according to the error, no?

Right, that's the question of what is `pydantic-factories` doing underneath when it tries to create an instance of a modal which has a relation field... There might be some bug somewhere deeper in the code, but I don't feel too comfortable lurking in there now, I don't know `ormar` nor `pydantic-factories` all that well—just learning the entire ecosystem yet :)

> Also, why are you typing as optional dict or person? Why not simply person?

Saw such type in `ormar` documentation and though it was the right way to type it. As mentioned earlier, I'm just getting to know FastAPI/ormar/pydantic ecosystem.

Reply:
BTW updated pytest to 7.0.1 to fix it not working on py >= 3.9